src/game.cpp

```diff
@@ -7247,11 +7259,13 @@ void game::zones_manager()
 cata::optional<tripoint> game::look_around()
 {
     tripoint center = u.pos() + u.view_offset;
-    return look_around( catacurses::window(), center, center, false, false );
+    cata::optional<peek_action_opt> peek_action = cata::nullopt;
+    return look_around( catacurses::window(), center, center, false, false, peek_action );
 }
 
 cata::optional<tripoint> game::look_around( catacurses::window w_info, tripoint &center,
-        const tripoint start_point, bool has_first_point, bool select_zone )
+        const tripoint start_point, bool has_first_point, bool select_zone,
+        cata::optional<peek_action_opt> &peek_action )
```

Comment:
Please don't make output reference parameters. The return value is for the output of the function, parameters are for the input. Make a new type containing the position and the action and return that (wrapped into `optional`).

Reply:
I updated this to return a new type and axed the output reference parameter as suggested. However, I made the return type non-optional, but the members of that new type (position and action) both are--it seemed more obtuse to work with if the return type was also wrapped in optional, since it should always be returning that object, but different properties may or may not be set  depending on the path through the function. Does that make sense?